File: utils/configloader.js

var _ = require('underscore');
var ArgumentParser = require('argparse').ArgumentParser;
var path = require('path');
var pkg = require('../package.json');

var parser = new ArgumentParser({
  version: pkg.version,
  addHelp: true,
  description: pkg.description
});

parser.addArgument(['-c','--config'], {
	help: 'Specify configuration file to use',
	dest: 'config',
	defaultValue: path.join(__dirname, '..', 'config.js')
});

parser.addArgument(['-r','--no-request-log'], {
	help: 'Disable HTTP Request Logging',
	dest: 'log_requests',
	action: 'storeFalse',
	defaultValue: true
});

exports.loadConfig = function() {
	var args = parser.parseArgs();	
	var config = require(args['config']).config;
	return _.extend(config, args);
}

File: rev/utils/configloader.py

import os
import argparse

import rev
from rev.core import PKG_NAME, PKG_VERSION

aparser = argparse.ArgumentParser(
                    description='{} v{}'.format(PKG_NAME, PKG_VERSION),
                    add_help=True,
)

aparser.add_argument('-c', '--config', 
            help='Specify configuration file to use',
            dest='configfile',
            default=os.path.join(os.getcwd(), 'revserver.conf'))

aparser.add_argument('-i', '--install',
            help='Specify a comma-seperated list of modules to install',
            dest='modules_to_install',
            default=None)

aparser.add_argument('-u', '--update',
            help='Specify a comma-seperated list of modules to update',
            dest='modules_to_update',
            default=None)

aparser.add_argument('-r', '--remove',
            help='Specify a comma-seperated list of modules to remove',
            dest='modules_to_remove',
            default=None)

def load_config():
    
    config = {}
    
    args = aparser.parse_args()
    
    if args.configfile:
        rev.log.info('Config File: %s', args.configfile)
        exec(open(args.configfile).read(), {}, config)
    
    config.update(vars(args))

    return config
